src/main/flight/imu.c

```diff
@@ -541,18 +567,31 @@ static void imuCalculateEstimatedAttitude(timeUs_t currentTimeUs)
         useMag = true;
     }
 #endif
+
 #if defined(USE_GPS)
-    if (!useMag && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat > GPS_MIN_SAT_COUNT && gpsSol.groundSpeed >= GPS_COG_MIN_GROUNDSPEED) {
-        // Use GPS course over ground to correct attitude.values.yaw
-        const float imuYawGroundspeed = fminf (gpsSol.groundSpeed / GPS_COG_MAX_GROUNDSPEED, 2.0f);
-        courseOverGround = DECIDEGREES_TO_RADIANS(gpsSol.groundCourse);
-        cogYawGain = (FLIGHT_MODE(GPS_RESCUE_MODE)) ? gpsRescueGetImuYawCogGain() : imuYawGroundspeed;
-        // normally update yaw heading with GPS data, but when in a Rescue, modify the IMU yaw gain dynamically
-        if (shouldInitializeGPSHeading()) {
-            // Reset our reference and reinitialize quaternion.
-            // shouldInitializeGPSHeading() returns true only once.
+    if (!useMag && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat > GPS_MIN_SAT_COUNT) {
+        static bool gpsHeadingInitialized = false;
+        if (gpsHeadingInitialized) {
+            courseOverGround = DECIDEGREES_TO_RADIANS(gpsSol.groundCourse);
+            // used in imuMahonyAHRSupdate()
+            if (FLIGHT_MODE(GPS_RESCUE_MODE)) {
+                // GPS_Rescue adjusts gpsGroundspeedGain during a rescue in a range 0 - 4.5, depending on GPS Rescue state and groundspeed relative to speed to home.
+                gpsGroundspeedGain = gpsRescueGetImuYawCogGain();
+            } else {
+                // in normal flight, IMU should:
+                // - heavily average GPS heading values at low speed, since they are random, almost
+                // - respond more quickly at higher speeds.
+                // GPS typically returns quite good heading estimates at or above 0.5- 1.0 m/s, quite solid by 2m/s
+                // gpsGroundspeedGain will be 0 at 0.0m/s, rising slowly towards 1.0 at 1.0 m/s, and reaching max of 10.0 at 10m/s
```

Comment:
It is easier to reason about IMU when gain is between 0 and 1.

Reply:
Yes, however this factor requires stronger relative weighting than the others, since in flat stable forward flight, when the other factors are both 1.0, GPS groundspeed is a crucial parameter for the accuracy of the GPS heading value.

Reply:
Ok. 

Guessing a bit, for reference only: 

Ideally, overall IMU gain would be close to 1 (and possibly removed completely) - minimizing (small, linear region) error in single step. Then different error sources would use gain associated with desired time constants. Then it would be a bit easier to reason about gain used.

Reply:
Yes it is very complex.  Each axis must, I think, have different weighting.